packages/vite/src/node/plugins/clientInjections.ts

```diff
@@ -35,6 +35,7 @@ export function clientInjectionsPlugin(config: ResolvedConfig): Plugin {
       const timeout = hmrConfig?.timeout || 30000
       const overlay = hmrConfig?.overlay !== false
       const isHmrServerSpecified = !!hmrConfig?.server
+      const isHmrTsConfig = config.configFile?.endsWith('.ts') || false
```

Comment:
Perhaps we can call `path.basename` instead so we cover custom named configs?

Reply:
Can you provide an example of what you mean by `custom named configs` ? I don't know very well.

Reply:
The user could run Vite like `pnpm dev --config vite.custom.js` instead, so the `basename` would be different.

Reply:
Oh, I see, I really didn't think about the situation, thanks!

Reply:
Something seems to go wrong after I updated. 🤔

Reply:
Our CI cache is full and GitHub is having issues evicting it automatically lately. It should be fine.